src/main/java/io/axoniq/axonserver/connector/impl/ControlChannelImpl.java

```diff
@@ -187,6 +190,12 @@ public synchronized void reconnect() {
 
     private void handleDisconnect(Throwable cause) {
         heartbeatMonitor.pause();
+        Status status = Status.fromThrowable(cause);
+        if (status == Status.UNAVAILABLE && channel.isReady()) {
+            // if the server is unavailable, we must trigger a reconnect
+            logger.info("Upstream unavailable. Forcing new connection.");
+            reconnectHandler.run();
+        }
         scheduleReconnect(cause);
```

Comment:
this should be in an ELSE statement

Reply:
This comment has been discussed in a video meeting and deemed to not cause any issues. 
The invocation of `scheduleReconnect(cause)` can introduce a double invocation of the reconnect, but it does not cause any issues whatsoever